### PR TITLE
fix: Fixes button group item states

### PR DESCRIPTION
**Pull request by pan-kot**
### Description

The PR includes the following fixes:
1. Menu items can be set as loading or disabled
2. Popover feedback can be shown for disabled and loading items

Related links, issue #, if available: n/a

### How has this been tested?

* New unit tests
* Improved test page and adjusted integ tests

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/button-group/menu-dropdown-item.tsx [hunk 1]
```diff
@@ -60,6 +57,9 @@ const MenuDropdownItem = React.forwardRef(
               ariaExpanded={ariaExpanded}
               className={clsx(testUtilStyles.item, testUtilsClass)}
               iconName="ellipsis"
+              loading={item.loading}
+              loadingText={item.loadingText}
+              disabled={item.disabled}
```
Comment (pan-kot) [inline src/button-group/menu-dropdown-item.tsx:62]: I created a new unit test for this change but it is also covered with item-permutations. Currently the menu item is rendered the same even when it is set as loading or disabled.

File: pages/button-group/test.page.tsx [hunk 2]
```diff
@@ -3,177 +3,239 @@
 
 import React, { useContext, useState } from 'react';
 
-import Box from '~components/box';
-import Button from '~components/button';
-import ButtonGroup, { ButtonGroupProps } from '~components/button-group';
-import SpaceBetween from '~components/space-between';
-import StatusIndicator from '~components/status-indicator';
+import { Box, Button, ButtonGroup, ButtonGroupProps, Header, SpaceBetween, StatusIndicator } from '~components';
```
Comment (orangevolon) [inline pages/button-group/test.page.tsx:6]: Aren't we encouraging the direct imports because of the treeshaking? If so, wouldn't it be better to keep it as before since our code base is being used as an example.

File: pages/button-group/test.page.tsx [hunk 3]
```diff
@@ -3,177 +3,239 @@
...
-  const onFocusOnMoreActionsButtonClick = () => {
-    ref.current?.focus('more-actions');
+    switch (detail.id) {
+      case 'like':
+      case 'dislike':
+        return syncAction(() => setFeedback(prev => (prev !== detail.id ? (detail.id as 'like' | 'dislike') : 'none')));
+      case 'favorite':
+        return asyncAction(() => setFavorite(prev => !prev));
+      case 'send':
+        return syncAction(() => setCanSend(false));
```
Comment (orangevolon) [inline pages/button-group/test.page.tsx:198]: Do we want the button to disappear after clicking the send, or should we show another state of the same button?
| Before click | After click |
| --- | --- |
| image | image |

File: src/button-group/__integ__/button-group.test.ts [hunk 4]
```diff
@@ -119,7 +120,7 @@ test(
     await page.keys(['Tab']);
     await expect(page.isFocused(likeButton.toSelector())).resolves.toBe(true);
 
-    await page.keys(['ArrowRight', 'ArrowRight']);
+    await page.keys(['ArrowRight', 'ArrowRight', 'ArrowRight', 'ArrowRight']);
```
Comment (orangevolon) [inline src/button-group/__integ__/button-group.test.ts:123]: np: since your already used `range()` in line `55` maybe we can repeat it here too?

File: pages/button-group/test.page.tsx (hunk 2 above)
Comment (pan-kot) [reply pages/button-group/test.page.tsx:6]: If we want to enforce this I think we should use an ESLint rule. Rn there are different import styles used across dev pages.

File: pages/button-group/test.page.tsx (hunk 3 above)
Comment (pan-kot) [reply pages/button-group/test.page.tsx:198]: The demo features two different behaviours. For the send button we hide it and for the redo button we make it disabled. I think the disappearing button is a very corner case but I think it is still worths having for completeness.

File: src/button-group/__integ__/button-group.test.ts (hunk 4 above)
Comment (pan-kot) [reply src/button-group/__integ__/button-group.test.ts:123]: I think here it is alright this way because 4 actions is easy to count anyways. I'd use range from 5-6 repeated actions personally.

**Review by orangevolon — APPROVED** [review]
